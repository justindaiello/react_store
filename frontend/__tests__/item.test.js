import ItemComponent from '../components/Item';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json'

const fakeItem = {
  id: 'Fake Item Id',
  title: 'Super Cool Item',
  price: 12500,
  description: 'This is a cool fake item for testing.',
  image: 'fakeItem.jpg',
  largeImage: 'largeFakeItem.jpg',
}

describe('<Item/>', () => {
  it('renders and matches the snapshot', () => {
    const wrapper = shallow(<ItemComponent item={fakeItem}/>);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  // it('Renders the image propery', () => {
  //   const wrapper = shallow(<ItemComponent item={fakeItem} />);
  //   const img = wrapper.find('img')
  //   expect(img.props().src).toBe(fakeItem.image);    
  //   expect(img.props().alt).toBe(fakeItem.title);   
  // })

  // it('Renders price tag and title properly', () => {
  //   const wrapper = shallow(<ItemComponent item={fakeItem} />);
  //   // console.log(wrapper.debug());  
  //   expect(wrapper.find('a').text()).toBe('Super Cool Item - $125'); 
  // })

  // it('Renders out buttons properly', () => {
  //   const wrapper = shallow(<ItemComponent item={fakeItem} />);
  //   const buttonList = wrapper.find('.buttonList');
  //   expect(buttonList.children()).toHaveLength(3);
  //   expect(buttonList.find('Link').exists()).toBe(true);
  //   expect(buttonList.find('AddToCart').exists()).toBe(true);
  //   expect(buttonList.find('DeleteItem').exists()).toBe(true);
  // })
})