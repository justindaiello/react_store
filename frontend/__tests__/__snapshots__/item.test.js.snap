// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Item/> renders and matches the snapshot 1`] = `
<ItemStyles__Item>
  <img
    alt="Super Cool Item"
    src="fakeItem.jpg"
  />
  <TitleStyles__Title>
    <Link
      href={
        Object {
          "pathname": "/item",
          "query": Object {
            "id": "Fake Item Id",
          },
        }
      }
    >
      <a>
        Super Cool Item
         - 
        $125
      </a>
    </Link>
  </TitleStyles__Title>
  <p>
    This is a cool fake item for testing.
  </p>
  <div
    className="buttonList"
  >
    <Link
      href={
        Object {
          "pathname": "update",
          "query": Object {
            "id": "Fake Item Id",
          },
        }
      }
    >
      <button>
        Edit ✏️
      </button>
    </Link>
    <AddToCart
      id="Fake Item Id"
    />
    <DeleteItem
      id="Fake Item Id"
    />
  </div>
</ItemStyles__Item>
`;
